mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceCluster.java

```diff
@@ -77,6 +79,12 @@ public interface ResourceCluster extends ResourceClusterGateway {
 
     CompletableFuture<TaskExecutorStatus> getTaskExecutorState(TaskExecutorID taskExecutorID);
 
+    CompletableFuture<Void> disabledTaskExecutorsAre(Map<String, String> attributes, Instant expiry);
```

Comment:
One thing I think requires further thinking is error/failure in upgrade workflow and what happens to this expiration. Othere then that the API lgtm.

Reply:
I expect this expiration to be set to a high value, such as a day after the command has been issued.